src/core/tsi/ssl_transport_security.c

```diff
@@ -1085,6 +1147,12 @@ static tsi_result create_tsi_ssl_handshaker(SSL_CTX *ctx, int is_client,
   impl->from_ssl = from_ssl;
   impl->result = TSI_HANDSHAKE_IN_PROGRESS;
   impl->base.vtable = &handshaker_vtable;
+
+  if (factory != NULL) {
+    impl->factory_ref = factory;
```

Comment:
with the suggested changes, you can just do:
`impl->factory_ref = tsi_ssl_handshaker_factory_ref(factory);`